mochify/lib/poll-events.js

```diff
@@ -3,22 +3,28 @@
 exports.pollEvents = pollEvents;
 
 async function pollEvents(driver, emit) {
-  for (;;) {
-    const events = await driver.evaluateReturn('mocha.mochify_pollEvents()');
-    if (!events) {
-      continue;
-    }
-    for (const [event, data] of events) {
-      if (event === 'mochify.callback') {
-        return data.code || 0; // stop polling
+  let interval;
+  const result = await new Promise((resolve) => {
+    interval = setInterval(async () => {
+      const events = await driver.evaluateReturn('mocha.mochify_pollEvents()');
+      if (!events) {
+        return;
       }
-      if (event === 'mochify.coverage') {
-        global.__coverage__ = data;
-      } else if (event.startsWith('console.')) {
-        console[event.substring(8)](...data);
-      } else {
-        emit(event, data);
+      for (const [event, data] of events) {
+        if (event === 'mochify.callback') {
+          resolve(data.code || 0);
+          return; // stop polling
+        }
+        if (event === 'mochify.coverage') {
+          global.__coverage__ = data;
+        } else if (event.startsWith('console.')) {
+          console[event.substring(8)](...data);
+        } else {
+          emit(event, data);
+        }
       }
-    }
-  }
+    }, 1);
```

Comment:
I guess this can lead to concurrent polling requests. Can you change it to poll, and then run the next poll slightly deferred? Maybe `setImmediate`?

Reply:
Yeah, sounds less risky to do it that way. Updated.